pymc4/distributions/continuous.py

```diff
@@ -1174,6 +1177,84 @@ def upper_limit(self):
         return self.conditions["high"]
 
 
+class Flat(ContinuousDistribution):
+    r"""A uniform distribution with support :math:`(-\inf, \inf)`.
+    Used as a uninformative log-likelihood that returns
+    zeros regardless of the passed values.
+    """
+
+    def __init__(self, name, **kwargs):
+        super().__init__(name, **kwargs)
+
+    @staticmethod
+    def _init_distribution(conditions):
+        return tfd.Uniform(low=-np.inf, high=np.inf)
+
+    def log_prob(self, value):
+        # convert the value to tensor
+        value = tf.convert_to_tensor(value)
+        expected = tf.zeros(self._distribution.batch_shape + self._distribution.event_shape)
+        # check if the event shape matches
+        if (
+            len(self._distribution.event_shape)
+            and value.shape[-len(self._distribution.event_shape) :]
+            != self._distribution.event_shape
+        ):
+            raise ValueError("values not consistent with the event shape of distribution")
+        # broadcast expected to shape of value
+        try:
+            expected = tf.broadcast_to(expected, value.shape)
+        except tf.python.framework.errors_impl.InvalidArgumentError:
+            raise ValueError("value can't be broadcasted to expected shape")
+
+        return tf.reduce_sum(expected, axis=range(-len(self._distribution.event_shape), 0))
+
+    def sample(self, shape=(), seed=None):
+        """Raises ValueError as it is not possible to sample
+        from flat distribution.
+        """
+        raise TypeError("cannot sample from a flat distribution")
+
+
+class HalfFlat(PositiveContinuousDistribution):
+    r"""Improper flat priors over positive reals."""
+
+    def __init__(self, name, **kwargs):
+        super().__init__(name, **kwargs)
+
+    @staticmethod
+    def _init_distribution(conditions):
+        return tfd.Uniform(low=0.0, high=np.inf)
+
+    def log_prob(self, value):
+        # convert the value to tensor
+        value = tf.convert_to_tensor(value)
+        value = tf.where(value > 0, x=0.0, y=-np.inf)
+        expected = tf.zeros(self._distribution.batch_shape + self._distribution.event_shape)
+        # check if the event shape matches
+        if (
+            len(self._distribution.event_shape)
+            and value.shape[-len(self._distribution.event_shape) :]
+            != self._distribution.event_shape
+        ):
+            raise ValueError("values not consistent with the event shape of distribution")
+        # broadcast expected to shape of value
+        try:
+            expected = tf.broadcast_to(expected, value.shape) + value
+        except tf.python.framework.errors_impl.InvalidArgumentError:
+            raise ValueError("value can't be broadcasted to expected shape")
```

Comment:
The same comment as for the `Flat` distribution, we need a test that raises this exception